### PR TITLE
MAKE-339: add time info to job interface

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: job/base.go [hunk 1]
```diff
@@ -178,3 +179,31 @@ func (b *Base) SetStatus(s amboy.JobStatusInfo) {
...
+}
+
+// UpdateTimeInfo updates the stored value of time in the job, but
+// does *not* modify fields that are unset in the input document.
+func (b *Base) UpdateTimeInfo(i amboy.JobTimeInfo) {
+	b.mutex.Lock()
+	defer b.mutex.Unlock()
+
+	if !i.StartTime.IsZero() {
+		b.TaskTimeInfo.StarTime = i.StartTime
```
Comment (bsamek) [inline job/base.go]: typo StartTime (this won't compile though?)

File: pool/group.go [hunk 2]
```diff
@@ -205,7 +206,13 @@ func groupWorker(ctx context.Context, wg *sync.WaitGroup, name string, work <-ch
 		case <-ctx.Done():
 			return
 		default:
+			ti := amboy.JobTimeInfo{
+				StarAt: time.Now(),
```
Comment (bsamek) [inline pool/group.go]: StartTime?

File: pool/local.go [hunk 3]
```diff
@@ -121,7 +122,14 @@ func worker(ctx context.Context, jobs <-chan amboy.Job, q amboy.Queue, wg *sync.
 				continue
 			}
 
+			ti := amboy.JobTimeInfo{
+				StarAt: time.Now(),
```
Comment (bsamek) [inline pool/local.go]: StartTime

File: pool/rate_limiting.go [hunk 4]
```diff
@@ -116,7 +116,14 @@ func (p *simpleRateLimited) worker(ctx context.Context, jobs <-chan amboy.Job) {
 			case <-ctx.Done():
 				return
 			case job := <-jobs:
+				ti := amboy.JobTimeInfo{
+					StarAt: time.Now(),
```
Comment (bsamek) [inline pool/rate_limiting.go]: StartTime

File: pool/group.go [hunk 5]
```diff
@@ -205,7 +206,13 @@ func groupWorker(ctx context.Context, wg *sync.WaitGroup, name string, work <-ch
...
+			}
 			unit.j.Run()
+			ti.EndAt = time.Now()
```
Comment (bsamek) [inline pool/group.go]: EndTime?

File: pool/local.go [hunk 6]
```diff
@@ -121,7 +122,14 @@ func worker(ctx context.Context, jobs <-chan amboy.Job, q amboy.Queue, wg *sync.
...
+			}
+
 			job.Run()
+			ti.EndAt = time.Now()
```
Comment (bsamek) [inline pool/local.go]: EndTime?

File: pool/rate_limiting.go [hunk 7]
```diff
@@ -116,7 +116,14 @@ func (p *simpleRateLimited) worker(ctx context.Context, jobs <-chan amboy.Job) {
...
+				}
+
 				job.Run()
+				ti.EndAt = time.Now()
```
Comment (bsamek) [inline pool/rate_limiting.go]: EndTime?

File: pool/rate_limiting_average.go [hunk 8]
```diff
@@ -172,7 +172,14 @@ func (p *ewmaRateLimiting) worker(ctx context.Context, jobs <-chan amboy.Job) {
 
 func (p *ewmaRateLimiting) runJob(ctx context.Context, j amboy.Job) time.Duration {
 	start := time.Now()
+	ti := amboy.JobTimeInfo{
+		StarAt: start,
+	}
+
 	j.Run()
+	ti.EndAt = time.Now()
```
Comment (bsamek) [inline pool/rate_limiting_average.go]: EndTime

File: pool/rate_limiting_average.go [hunk 9]
```diff
@@ -172,7 +172,14 @@ func (p *ewmaRateLimiting) worker(ctx context.Context, jobs <-chan amboy.Job) {
 
 func (p *ewmaRateLimiting) runJob(ctx context.Context, j amboy.Job) time.Duration {
 	start := time.Now()
+	ti := amboy.JobTimeInfo{
+		StarAt: start,
```
Comment (bsamek) [inline pool/rate_limiting_average.go]: StartTime

**Comment by bsamek** [comment]
Also, it doesn't look like PR testing is working with this repo. Are there sufficient permissions? And the last couple commits have broken tests.

**Comment by tychoish** [comment]
Fixed compile. I'll add some tests.

**Pull request merged**